src/node_crypto_clienthello.h

```diff
@@ -108,16 +108,16 @@ class ClientHelloParser {
   OnHelloCb onhello_cb_;
   OnEndCb onend_cb_;
   void* cb_arg_;
-  size_t frame_len_;
-  size_t body_offset_;
-  size_t extension_offset_;
+  size_t frame_len_ = 0;
+  size_t body_offset_ = 0;
+  size_t extension_offset_ = 0;
   uint8_t session_size_;
   const uint8_t* session_id_;
   uint16_t servername_size_;
   const uint8_t* servername_;
   uint8_t ocsp_request_;
```

Comment:
I think you also need/want to add default initializers here, for the 5 that are later applied to `ClientHello` directly?

Reply:
I've updated the PR and moved the default initializers from `ClientHello` to `ClientHelloParser`, which I believe is what was needed in the first place.

Thanks for catching that. I'm not very familiar with C++ and appreciate the keen eye!

Reply:
@theholla Just for context – I don’t think there was anything wrong with adding the initializers to `ClientHello`, as long as `ClientHelloParser` has all initializers that were previously in its own constructor :)

Reply:
That's good to know, thanks! While you're here, I wonder if you see any issues with the two fields that are listed as one type in one class, and another in the next? Both `servername_size` and `ocsp_request` are listed as different types between `ClientHello` and `ClientHelloParser`.
```
class ClientHelloParser {
  class ClientHello {
    private:
      uint8_t servername_size_;
      int ocsp_request_;
  };
  private:
    uint16_t servername_size_ = 0;
    uint8_t ocsp_request_ = 0;
}
```
This predated the PR but stood out to me as a possible issue.

Reply:
Puh … that’s a *good* question. 

For `ocsp_request_`, I searched for instances of that in the source code, and it seems like the only values are `0` and `1` – I think you could change the type in one so it matches the other, if you like. (I could totally be wrong about this being a “boolean“, though…)

For `servername_size_`, I’m not sure. It *looks* to me like we wouldn’t do anything wrong by making both fields `uint16_t`s, and remove the `static_cast` when converting …

Either way, this might be better answered by @nodejs/crypto experts?